engineccl: failure in BenchmarkTimeBoundIterate

On seemingly unrelated PR ([here](https://teamcity.cockroachdb.com/viewLog.html?buildId=11699672&buildTypeId=Cockroach_Ci_Tests_Bench&tab=buildLog#_state=218,3070)):
```
Bench //pkg/ccl/storageccl/engineccl:engineccl_test] --- FAIL: BenchmarkTimeBoundIterate/LoadFactor=0.50/TimeBoundIterator
[01:17:40][Bench //pkg/ccl/storageccl/engineccl:engineccl_test]     test_fixtures.go:90: using existing fixture "mvcc_data_vv1000023.1-22_100000_100_10_512" in "/home/roach/.cache/crdb-test-fixtures/mvcc_data_vv1000023.1-22_100000_100_10_512"
[01:17:40][Bench //pkg/ccl/storageccl/engineccl:engineccl_test]     bench_test.go:167: expected at least 50000 keys, but got 49960
```


Jira issue: CRDB-31361

--- a/pkg/ccl/storageccl/engineccl/BUILD.bazel
+++ b/pkg/ccl/storageccl/engineccl/BUILD.bazel
@@ -51,7 +51,6 @@ go_test(
         "//pkg/storage/enginepb",
         "//pkg/testutils",
         "//pkg/testutils/datapathutils",
-        "//pkg/testutils/skip",
         "//pkg/testutils/storageutils",
         "//pkg/testutils/testfixtures",
         "//pkg/util/encoding",

--- a/pkg/ccl/storageccl/engineccl/bench_test.go
+++ b/pkg/ccl/storageccl/engineccl/bench_test.go
@@ -21,7 +21,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
 	"github.com/cockroachdb/cockroach/pkg/storage"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testfixtures"
 	"github.com/cockroachdb/cockroach/pkg/util/encoding"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
@@ -29,20 +28,25 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/randutil"
 )
 
-// loadTestData writes numKeys keys in numBatches separate batches. Keys are
-// written in order. Every key in a given batch has the same MVCC timestamp;
-// batch timestamps start at batchTimeSpan and increase in intervals of
-// batchTimeSpan.
+// loadTestData writes numKeys keys in numBatches separate batches. Each key is
+// in the form 'key-XXXXX' where XXXXXX is an ascending-encoded uvarint of the
+// key index. Keys are written in order. Every batch is assigned a minimum MVCC
+// timestamp T_min. All keys within the batch have timestamps in the interval
+// [T_min,T_min+batchTimeSpan). Each successive batch has a T_min that's
+// batchTimeSpan higher than the previous. Thus batches are disjoint in both
+// user key space and MVCC timestamps.
 //
-// Importantly, writing keys in order convinces RocksDB to output one SST per
-// batch, where each SST contains keys of only one timestamp. E.g., writing A,B
-// at t0 and C at t1 will create two SSTs: one for A,B that only contains keys
-// at t0, and one for C that only contains keys at t1. Conversely, writing A, C
-// at t0 and B at t1 would create just one SST that contained A,B,C (due to an
-// immediate compaction).
+// After each batch is committed, a memtable flush is forced to compel Pebble
+// into separating batches into separate sstables where each sstable contains
+// keys only for a single batch (and a single `batchTimeSpan` time window).
 //
-// The creation of the database is time consuming, so the caller can choose
-// whether to use a temporary or permanent location.
+// The creation of the database is mildly time consuming, so it's cached as a
+// test fixture (see testfixtures.ReuseOrGenerate).
+//
+// TODO(jackson): This test was initially written when time-bound iteration
+// could only optimize iteration by skipping entire sstables. With the
+// introduction of block-property filters in 22.1, this is no longer the case.
+// The test could be adapted to test more granular iteration, if useful.
 func loadTestData(
 	tb testing.TB, dirPrefix string, numKeys, numBatches, batchTimeSpan, valueBytes int,
 ) storage.Engine {
@@ -69,9 +73,9 @@ func loadTestData(
 			keys[i] = roachpb.Key(encoding.EncodeUvarintAscending([]byte("key-"), uint64(i)))
 		}
 
-		sstTimestamps := make([]int64, numBatches)
-		for i := 0; i < len(sstTimestamps); i++ {
-			sstTimestamps[i] = int64((i + 1) * batchTimeSpan)
+		minSStableTimestamps := make([]int64, numBatches)
+		for i := 0; i < len(minSStableTimestamps); i++ {
+			minSStableTimestamps[i] = int64(i * batchTimeSpan)
 		}
 
 		var batch storage.Batch
@@ -89,7 +93,7 @@ func loadTestData(
 					}
 				}
 				batch = eng.NewBatch()
-				minWallTime = sstTimestamps[i/scaled]
+				minWallTime = minSStableTimestamps[i/scaled]
 			}
 			timestamp := hlc.Timestamp{WallTime: minWallTime + rand.Int63n(int64(batchTimeSpan))}
 			value := roachpb.MakeValueFromBytes(randutil.RandBytes(rng, valueBytes))
@@ -137,7 +141,7 @@ func runIterate(
 
 	// Store the database in this directory so we don't have to regenerate it on
 	// each benchmark run.
-	eng := loadTestData(b, "mvcc_data", numKeys, numBatches, batchTimeSpan, valueBytes)
+	eng := loadTestData(b, "mvcc_data_v2", numKeys, numBatches, batchTimeSpan, valueBytes)
 	defer eng.Close()
 
 	b.SetBytes(int64(numKeys * valueBytes))
@@ -173,7 +177,6 @@ func runIterate(
 }
 
 func BenchmarkTimeBoundIterate(b *testing.B) {
-	skip.WithIssue(b, 110299)
 	for _, loadFactor := range []float32{1.0, 0.5, 0.1, 0.05, 0.0} {
 		b.Run(fmt.Sprintf("LoadFactor=%.2f", loadFactor), func(b *testing.B) {
 			b.Run("NormalIterator", func(b *testing.B) {

